core/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneOptimized.java

```diff
@@ -338,7 +339,16 @@ public CanonicalGene getNonAmbiguousGene(String geneId) {
      * @param chr chromosome
      * @return a gene that can be non-ambiguously determined, or null if cannot.
      */
-    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) {
+    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) { return getNonAmbiguousGene(geneId, chr, null); }
+
+        /**
+         * Look for gene that can be non-ambiguously determined.
+         * @param geneId an Entrez Gene ID or HUGO symbol or gene alias
+         * @param chr chromosome
+         * @param pMonitor for import warnings
+         * @return a gene that can be non-ambiguously determined, or null if cannot.
+         */
+    public CanonicalGene getNonAmbiguousGene(String geneId, String chr, ProgressMonitor pMonitor) {
```

Comment:
I agree that singleton would be useful. The getUnambiguousGeneName is also called on tomcat so we have to make sure that case is also handled - only print the warning to the log


________________________________
From: JianJiong Gao [notifications@github.com]
Sent: Tuesday, February 02, 2016 8:11 PM
To: cBioPortal/cbioportal
Cc: Jan Hudecek
Subject: Re: [cbioportal] Added sorting of warnings during import and displaying them with counts (#807)


In core/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneOptimized.java<https://github.com/cBioPortal/cbioportal/pull/807#discussion_r51618365>:

> @@ -338,7 +339,16 @@ public CanonicalGene getNonAmbiguousGene(String geneId) {
>       * @param chr chromosome
>       * @return a gene that can be non-ambiguously determined, or null if cannot.
>       */
> -    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) {
> +    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) { return getNonAmbiguousGene(geneId, chr, null); }
> +
> +        /**
> +         * Look for gene that can be non-ambiguously determined.
> +         * @param geneId an Entrez Gene ID or HUGO symbol or gene alias
> +         * @param chr chromosome
> +         * @param pMonitor for import warnings
> +         * @return a gene that can be non-ambiguously determined, or null if cannot.
> +         */
> +    public CanonicalGene getNonAmbiguousGene(String geneId, String chr, ProgressMonitor pMonitor) {


I don't like to put pMoniter in the function signature. Maybe we should have a ProgressMonitor singleton class that script and dao classes can call? And the singleton class is only initialized when importing (not the web code).

�
Reply to this email directly or view it on GitHub<https://github.com/cBioPortal/cbioportal/pull/807/files#r51618365>.